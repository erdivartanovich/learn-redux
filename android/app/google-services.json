{
  "project_info": {
    "project_number": "297640754642",
    "firebase_url": "https://todo-79a2c.firebaseio.com",
    "project_id": "todo-79a2c",
    "storage_bucket": "todo-79a2c.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:297640754642:android:f1a54d186d74012a",
        "android_client_info": {
          "package_name": "com.refactory.redux"
        }
      },
      "oauth_client": [
        {
          "client_id": "297640754642-d707lgfec02tfuvpi9nk9p12mv0hpnc5.apps.googleusercontent.com",
          "client_type": 1,
          "android_info": {
            "package_name": "com.refactory.redux",
            "certificate_hash": "0D7485BCAA2BEC870758E09DAE49C9732B2BA5B6"
          }
        },
        {
          "client_id": "297640754642-opocmgn7if7l7vskrhn3r3fa7jbvrt34.apps.googleusercontent.com",
          "client_type": 3
        }
      ],
      "api_key": [
        {
          "current_key": "AIzaSyBnszvi8Rs5T_mZJA774g0MQM0pC3mG4lQ"
        }
      ],
      "services": {
        "analytics_service": {
          "status": 1
        },
        "appinvite_service": {
          "status": 2,
          "other_platform_oauth_client": [
            {
              "client_id": "297640754642-rrh0p6h4mv3bmv8676i4blb7clp2qs89.apps.googleusercontent.com",
              "client_type": 2,
              "ios_info": {
                "bundle_id": "com.refactory.redux"
              }
            },
            {
              "client_id": "297640754642-opocmgn7if7l7vskrhn3r3fa7jbvrt34.apps.googleusercontent.com",
              "client_type": 3
            }
          ]
        },
        "ads_service": {
          "status": 2
        }
      }
    }
  ],
  "configuration_version": "1"
}