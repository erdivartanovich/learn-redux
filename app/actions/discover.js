export const SET_VENUE_TAG = 'SET_VENUE_TAG';
export const SET_VENUE_NEARBY = 'SET_VENUE_NEARBY';
export const SET_VENUE_CITY = 'SET_VENUE_CITY';

export function setVenueTag(tag) {
    return {type: SET_VENUE_TAG, tag};
}

export function setVenueNearby(is_nearby, coordinate) {
    return {type: SET_VENUE_NEARBY, is_nearby, coordinate};
}

export function setVenueCity(city) {
    return {type: SET_VENUE_CITY, city};
}
