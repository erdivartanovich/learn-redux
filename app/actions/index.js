/*
 * action types
 */

export const ADD_TODO = 'ADD_TODO'
export const TOGGLE_TODO = 'TOGGLE_TODO'
export const SET_VISIBILITY_FILTER = 'SET_VISIBILITY_FILTER'
export const UPDATE_TODO_TEXT = 'UPDATE_TODO_TEXT';
export const GET_TODO = 'GET_TODO';
export const POST_TODO = 'POST_TODO';
/*
 * other constants
 */

export const VisibilityFilters = {
  SHOW_ALL: 'SHOW_ALL',
  SHOW_COMPLETED: 'SHOW_COMPLETED',
  SHOW_ACTIVE: 'SHOW_ACTIVE'
}


export function updateText(text) {
    return {type: UPDATE_TODO_TEXT, text};
}
/*
 * action creators
 */

export function addTodo(text) {
  return { type: ADD_TODO, text }
}

export function toggleTodo(index) {
  return { type: TOGGLE_TODO, index }
}

export function setVisibilityFilter(filter) {
  return { type: SET_VISIBILITY_FILTER, filter }
}
