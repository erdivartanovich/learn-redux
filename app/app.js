
import React from 'react';
import {
  View,
} from 'react-native';
import DiscoverVenue from './containers/DiscoverVenue';
import FilterVenue from './containers/FilterVenue';

const App = () => (
  <View style={{ flex: 1, flexGrow: 1, paddingTop: 20 }}>
    <FilterVenue />
    <DiscoverVenue />
  </View>
);

export default App;
