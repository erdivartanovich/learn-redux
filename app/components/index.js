import Todo from './Todo';
import TodoList from './TodoList';

export default {
    Todo,
    TodoList
};
