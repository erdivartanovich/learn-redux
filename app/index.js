import React, { Component } from 'react';
import {Provider} from 'react-redux';
import {createStore, applyMiddleware, combineReducers, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import createLogger from 'redux-logger';
import todoApp from './reducers';
import App from './app'
const logger = createLogger();
let store = createStore(todoApp, applyMiddleware(thunkMiddleware));
const Main = (props) => (<Provider store={store}>
    <App/>
</Provider>);
export default Main
