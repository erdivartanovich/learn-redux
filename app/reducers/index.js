import { combineReducers } from 'redux';
import { ADD_TODO, TOGGLE_TODO, SET_VISIBILITY_FILTER, UPDATE_TODO_TEXT, VisibilityFilters } from '../actions';
import { SET_VENUE_TAG, SET_VENUE_NEARBY, SET_VENUE_CITY } from '../actions/discover';

const { SHOW_ALL } = VisibilityFilters;
function visibilityFilter(state = SHOW_ALL, action) {
  switch (action.type) {
    case SET_VISIBILITY_FILTER:
      return action.filter;
    default:
      return state;
  }
}

function todoText(state = '', action) {
  switch (action.type) {
    case UPDATE_TODO_TEXT:
      return action.text;
    default:
      return state;
  }
}

function todos(state = [], action) {
  switch (action.type) {
    case ADD_TODO:
      return [
        ...state, {
          title: action.text,
          id: new Date().getTime(),
          completed: false,
          synced: false,
        },
      ];
    case TOGGLE_TODO:
      return state.map((todo, index) => {
        if (index === action.index) {
          return Object.assign({}, todo, {
            completed: !todo.completed,
          });
        }
        return todo;
      });
    default:
      return state;
  }
}

function discover(state = { is_nearby: false, tag: '', coordinate: {}, city: '' }, action) {
  switch (action.type) {
    case SET_VENUE_TAG:
      return { ...state, tag: action.tag };
    case SET_VENUE_NEARBY:
      return { ...state, is_nearby: action.is_nearby, coordinate: action.coordinate };
    case SET_VENUE_CITY:
      return { ...state, city: action.city };
    default:
      return state;
  }
}


export default combineReducers({ visibilityFilter, todoText, todos, discover });
