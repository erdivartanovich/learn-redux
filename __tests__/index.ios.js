import 'react-native';
import React from 'react';
import todoApp from '../app/reducers';
import {createStore, applyMiddleware, combineReducers, compose} from 'redux';
let store = createStore(todoApp);
import {addTodo} from '../app/actions';
store.dispatch(addTodo('hello world'));
import Index from '../index.ios.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {

  const tree = renderer.create(
    <Index />
  );
  expect(tree).toMatchSnapshot();

});
