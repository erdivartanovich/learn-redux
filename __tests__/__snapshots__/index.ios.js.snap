exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "flex": 1,
        "marginBottom": 40,
        "marginTop": 40,
      }
    }>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }>
      <TextInput
        onChangeText={[Function]}
        placeholder="Input new todo"
        style={
          Object {
            "flex": 1,
          }
        } />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          Add New
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Empty Content
    </Text>
  </View>
</View>
`;
